uefi/src/system.rs

```diff
@@ -0,0 +1,94 @@
+//! TODO
+
+// TODO
+#![allow(clippy::missing_safety_doc)]
+
+use crate::proto::console::text;
+use crate::table::{cfg, Boot, Revision, SystemTable};
+use crate::CStr16;
+use core::ptr::{self, NonNull};
+use core::slice;
+use core::sync::atomic::{AtomicPtr, Ordering};
+
+static SYSTEM_TABLE: AtomicPtr<uefi_raw::table::system::SystemTable> =
+    AtomicPtr::new(ptr::null_mut());
+
+/// TODO
+pub unsafe fn set_system_table(system_table: *mut uefi_raw::table::system::SystemTable) {
+    SYSTEM_TABLE.store(system_table, Ordering::Release);
```

Comment:
nit: This should use a `Once`-like type in the background. It should not be possible to set this multiple times.

I think something like `AtomicOnce<uefi_raw::table::system::SystemTable>` would be cool